File: src/modes.js
import { CONTEXT_MODES, MVP_CONTEXT_MODES } from "./constants.js";
import { ERROR_CODES, contextError } from "./errors.js";

export function validateContextMode(contextMode) {
  const allowedModes = Object.values(CONTEXT_MODES);
  if (!allowedModes.includes(contextMode)) {
    throw contextError(ERROR_CODES.VALIDATION_ERROR, "contextMode is not recognized", {
      details: { field: "contextMode", allowedModes }
    });
  }

  if (!MVP_CONTEXT_MODES.includes(contextMode)) {
    throw contextError(ERROR_CODES.CONTEXT_MODE_UNSUPPORTED, "contextMode is not supported in MVP", {
      httpStatus: 422,
      details: { contextMode, supportedModes: MVP_CONTEXT_MODES }
    });
  }

  return contextMode;
}

export function isMvpContextMode(contextMode) {
  return MVP_CONTEXT_MODES.includes(contextMode);
}

File: src/ai_assist_context_service/modes.py
from .constants import CONTEXT_MODES, MVP_CONTEXT_MODES
from .errors import ERROR_CODES, context_error


def validate_context_mode(context_mode):
    allowed_modes = tuple(CONTEXT_MODES.values())
    if context_mode not in allowed_modes:
        raise context_error(
            ERROR_CODES["VALIDATION_ERROR"],
            "contextMode is not recognized",
            details={"field": "contextMode", "allowedModes": allowed_modes},
        )

    if context_mode not in MVP_CONTEXT_MODES:
        raise context_error(
            ERROR_CODES["UNSUPPORTED_CONTEXT_MODE"],
            f"{context_mode} is not supported for the MVP read path.",
            http_status=400,
            category="VALIDATION",
            target="contextMode",
            details={"contextMode": context_mode, "supportedModes": MVP_CONTEXT_MODES},
        )

    return context_mode


def is_mvp_context_mode(context_mode):
    return context_mode in MVP_CONTEXT_MODES
